### PR TITLE
Firewalld: Allow to modify the default zone

### PR DESCRIPTION
**Pull request by teclator**
Trello Card: https://trello.com/c/jL8fhEKw/1091-5-firewall-firewalld-is-the-new-susefirewall-autoyast-support

### CODE REVIEW

File: library/network/test/y2firewall/firewalld_test.rb [hunk 1]
```diff
@@ -179,4 +181,142 @@
...
+      allow(firewalld).to receive("api").and_return api
+    end
+
+    it "returns false if firewalld is not installed" do
+      allow(firewalld).to receive(:installed?).and_return(false)
+
+      expect(firewalld.read).to eq(false)
+    end
+
+    it "initializes @zones parsing the firewalld zones list" do
```
Comment (imobachgs) [inline library/network/test/y2firewall/firewalld_test.rb]: I would replace *@zones* by something like *the list of zones* (being a instance variable is an implementation detail).

File: library/network/test/y2firewall/firewalld_test.rb [hunk 2]
```diff
@@ -179,4 +181,142 @@
...
+      firewalld.read
+
+      external = firewalld.find_zone("external")
+      expect(external.ports).to eql(["5901/tcp", "5901/udp"])
+    end
+
+    it "initializes the @default_zone and @log_denied_packets with the firewalld config" do
```
Comment (imobachgs) [inline library/network/test/y2firewall/firewalld_test.rb]: Same than above.

File: library/network/test/y2firewall/firewalld_test.rb [hunk 3]
```diff
@@ -179,4 +181,142 @@
...
+        expect(firewalld.modified?).to eq(true)
+      end
+    end
+
+    context "when no attribute has been modifiede since read" do
+      it "returns false" do
+        firewalld.default_zone = "public"
+        expect(firewalld.modified?).to eq(false)
+      end
+
```
Comment (imobachgs) [inline library/network/test/y2firewall/firewalld_test.rb]: rubocop: blank line :)

**Comment by teclator** [comment]
@imobachgs fixed conflicts, updated based on CR and added export logic

File: library/network/test/y2firewall/firewalld_test.rb [hunk 4]
```diff
@@ -179,4 +181,183 @@
...
+      firewalld.read
+    end
+
+    it "returns a hash with the current firewalld config" do
+      config = firewalld.export
+      external = config["zones"].find { |z| z["name"] == "external" }
+
+      expect(config).to be_a(Hash)
+      expect(config["log_denied_packets"]).to eq(true)
+      expect(config["default_zone"]).to eq("work")
```
Comment (imobachgs) [inline library/network/test/y2firewall/firewalld_test.rb]: np: Why do you use `eq` here and `eql` in the line below?

Comment (teclator) [reply library/network/test/y2firewall/firewalld_test.rb]: Probably fast typing makes me omit the "l" and then yanked and pasted, was not intentional to use eq in this case so, will modify it, Thnx ;)

**Pull request merged**